src/HTTPClient.php

```diff
@@ -101,6 +113,33 @@ protected function doRequest(string $path, Message $request_class, string $reply
         return $obj;
     }
 
+    /**
+     * Make the request
+     * @param Client $client the http client
+     * @param string $method method of the call
+     * @param string $path path being called
+     * @param string $json request json
+     * @param float $timeout timeout in milliseconds
+     * @return mixed|ResponseInterface response json
+     * @throws SDKException
+     */
+    private function call(Client $client, string $method, string $path, string $json, float $timeout) {
+        try {
+            return $client->request(
+                $method,
+                $this->buildURL($path),
+                [
+                    RequestOptions::BODY => $json,
+                    RequestOptions::TIMEOUT => $timeout / 1000 // seconds
+                ]
+            );
+        } catch (GuzzleException $e) {
+            // cURL doesn't return a 408 on timeouts, and also sets the code the 0... :
```

Comment:
`Vendasta\Vax\SDKException: Error calling /sso.v1.IdentityProvider/GetEntryURL (code: 0): cURL error 28: Operation timed out after 102 milliseconds with 0 out of 0 bytes received (see http://curl.haxx.se/libcurl/c/libcurl-errors.html)`

Is what we  get back from a request that timed out. Error 28 from that url is `CURLE_OPERATION_TIMEDOUT`.